packages/grafana-data/src/field/displayProcessor.ts

```diff
@@ -45,10 +45,23 @@ export function getDisplayProcessor(options?: DisplayProcessorOptions): DisplayP
 
   let unit = config.unit;
   let hasDateUnit = unit && (timeFormats[unit] || unit.startsWith('time:'));
+  let showMs = false;
 
   if (field.type === FieldType.time && !hasDateUnit) {
     unit = `dateTimeAsSystem`;
     hasDateUnit = true;
+    if (field.values && field.values.length > 1) {
```

Comment:
Is it possible this logic messes up the table panel? I have a table with a column of date times, all less than a minute apart.

I also have my date time format set to `GF_DATE_FORMATS_FULL_DATE=MMM Do, YYYY HH:mm:ss`

Yet, I see 
![StartTimesWithMsEvenThoughIJustWantS](https://user-images.githubusercontent.com/92043606/188943217-21a23bdd-5b30-48c7-b27e-dbf0b96628c6.png)

Reply:
> I have a table with a column of date times, all less than a minute apart.

the above doesnt look like less than a minute apart. i think what's happening here is that the data is not sorted ascending, and this PR assumes it is.

we can wrap the heuristic in `Math.abs()` so it works consistently for reverse-sorted data.

https://github.com/grafana/grafana/blob/5a3ae293cfad304e970435fcdd6fcab476979b9e/packages/grafana-data/src/field/displayProcessor.ts#L65

Reply:
You are correct. The timestamps are very obviously not a minute apart - oops! So I guess it's the sort. Generally seems a bit strange to have a heuristic for displaying the date format on tables. I would expect users to want to specify the display format exactly.

Reply:
it's a balancing act. we rely on the single displayProcessor that's created for each field in various contexts (legend, tooltip, axes ticks, barchart value rendering, stat value rendering, etc). it's difficult to accommodate all cases explicitly without exploding the options to provide fine-grained control over every context. i'll see if there's more we can do when the format is explicitly set, though.

Reply:
Yep, totally makes sense. This function has a large surface area. I appreciate you looking at it!

Reply:
https://github.com/grafana/grafana/pull/54923